### PR TITLE
COMP: fix signed/unsigned comparison introduced by pull request 5045.

### PR DESCRIPTION
**Pull request by Leengit**
Fixes warning for comparison of signed and unsigned integer.

### CODE REVIEW

File: Libs/MRML/Core/vtkMRMLMessageCollection.cxx [hunk 1]
```diff
@@ -63,7 +63,8 @@ vtkMRMLMessageCollection
 ::GetNumberOfMessagesOfType(unsigned long messageType) const
 {
   int response = 0;
-  for (int i = 0; i < this->Messages.size(); ++i)
+  const unsigned numberOfMessages = static_cast<unsigned>(this->Messages.size());
+  for (unsigned i = 0; i < numberOfMessages; ++i)
     {
     response += static_cast<int>(GetNthMessageType(i) == messageType);
```
Comment (lassoan) [inline Libs/MRML/Core/vtkMRMLMessageCollection.cxx:69]: Bool should not be casted to int. You can use the ternary operator or a simple `if` to increment by 1.

Comment (jcfr) [reply Libs/MRML/Core/vtkMRMLMessageCollection.cxx:69]: I suggest something like this:

```cpp
  int response = 0;
  for (int i = 0; i < static_cast<int>(this->Messages.size()); ++i)
    {
    if (this->GetNthMessageType(i) == messageType)
      {
      response++;
      }
    }
  return response;
```

**Comment by jcfr** [comment]
Consider also using a more descriptive commit message. Looking at the commit history may help.

File: Libs/MRML/Core/vtkMRMLMessageCollection.cxx (hunk 1 above)
Comment (Leengit) [reply Libs/MRML/Core/vtkMRMLMessageCollection.cxx:69]: Will do simple `if`.

**Comment by jcfr** [comment]
Thanks, I will integrate updating the commit message to the following:

```
COMP: Fix -Wsign-comp warning in vtkMRMLMessageCollection

This commit fixes a regression introduced in b30151532 (ENH: Display warnings
and errors in the Save Data dialog.)
```

**Comment by Leengit** [comment]
> Thanks, I will integrate updating the commit message to the following:
> 
> ```
> COMP: Fix -Wsign-comp warning in vtkMRMLMessageCollection
> 
> This commit fixes a regression introduced in b30151532 (ENH: Display warnings
> and errors in the Save Data dialog.)
> ```

Great!  Thank you.

**Pull request merged**